### PR TITLE
[pt] rule CONFUSÃO_MAIS_MAS_VÍRGULA improved

### PR DESCRIPTION
**Pull request by susanaboatto**
Hi all @St-ac-y @marcoagpinto 

I have improved the mais/mas confusion rule to fix some false alarms that were showing up. Please note that a big portion of the examples shown [here](https://internal1.languagetool.org/regression-tests/via-http/2022-07-06/pt-PT/result_grammar_CONFUSION_OF_MAS_MAIS[1].html) are valid corrections for the rule. I have added antipatterns for the ones that do not apply :)

If anyone wants to take a look at it, please let me know your thoughts.

### CODE REVIEW

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 1]
```diff
@@ -341,30 +341,129 @@ title="Easy editing stylesheet" ?>
...
+            <token postag='N.+' postag_regexp='yes'/>
+            <token>,</token>
+            <token>mais</token>
+            <token postag='N.+' postag_regexp='yes'/>
+            <example>Menos conversa, mais ação.</example>
+        </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token>mais</token>
+            <token regexp='no' min='0'>de</token>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:394]: I think `regexp="no"` is the default. You do not need to put that in.

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 2]
```diff
@@ -341,30 +341,129 @@ title="Easy editing stylesheet" ?>
...
+        </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token skip="-1">mais</token>
+            <token regexp='no' min='0'>ou</token>
+            <token>menos</token>
+            <example>Ele vai ganhar, mais voto ou menos voto.</example>
+        </antipattern>
+        <antipattern>
+            <token regexp='no' min='0'>ainda</token>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:413]: same

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 3]
```diff
@@ -341,30 +341,129 @@ title="Easy editing stylesheet" ?>
...
+            <token skip="-1">mais</token>
+            <token regexp='no' min='0'>do</token>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:415]: same

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml (hunk 1 above)
Comment (susanaboatto) [reply languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:394]: Thanks, I have fixed them all :)

**Review by St-ac-y — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 4]
```diff
@@ -417,18 +417,18 @@ title="Easy editing stylesheet" ?>
             <example>Este grupo, pouco conhecido, mais afasta do que atrai fiéis.</example>
         </antipattern>
         <antipattern>
-             <token regexp='yes'>'C.+|I.+|enfim|opa'</token>
+             <token postag="C.+|I.+|enfim|opa" postag_regexp="yes"/>
              <token>,</token>
              <token>mais</token>
              <token regexp='yes' min='0'>'N.+|PD.+|DD.+'</token>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:423]: `postag="'N.+|PD.+|DD.+'"`

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 5]
```diff
@@ -341,30 +341,129 @@ title="Easy editing stylesheet" ?>
...
+            <token>mais</token>
+            <token regexp='yes'>cedo|tarde|à|precisamente|frequentemente|recentemente|detalhadamente|adiante|acima|abaixo|atrás|ao|para|vale|nada|ou|devagar|rápido|um|uma|algum|alguma</token>
             <example>O local é perto do shopping, mais adiante.</example>
         </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token>mais</token>
+            <token regexp='no' min='0'>'VMP00SM'</token>
+            <token regexp='yes'>ainda|cedo|tarde|adiante|abaixo|</token>
+            <token regexp='yes'>.|,|C.+|SPS00</token>
```
[the remark this answers is not in the document]
Comment (danielnaber) [reply languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:458]: Also, POS tags and tokens cannot be in a single regex like this, should be something like `<token postag="SPS00">,</token>` (not sure of that makes sense, it means "," that has the "SPS00" tagging, so probably wouldn't match.)

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 6]
```diff
@@ -420,7 +420,7 @@ title="Easy editing stylesheet" ?>
              <token postag="C.+|I.+|enfim|opa" postag_regexp="yes"/>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:420]: again: `<token postag="C.+|I.+|enfim|opa" postag_regexp="yes"/>` does not seem to work
`C.+|I.+` seem to be postags and `enfim|opa` seem to be actual words.
they need to be  separated into two APs

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 7]
```diff
@@ -455,9 +455,17 @@ title="Easy editing stylesheet" ?>
...
             <token regexp='yes'>ainda|cedo|tarde|adiante|abaixo|</token>
-            <token regexp='yes'>\.|,|C.+|SPS00</token>
+            <token regexp='yes'>\.|,</token>
             <example>Já gostava de chocolate - e agora, mais ainda.</example>
         </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token>mais</token>
+            <token postag='VMP00SM' min='0'/>
+            <token regexp='yes'>ainda|cedo|tarde|adiante|abaixo|</token>
```
Comment (St-ac-y) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml:465]: delete the extra `|` at the end

**Review by St-ac-y — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml [hunk 8]
```diff
@@ -453,17 +453,11 @@ title="Easy editing stylesheet" ?>
...
             <token regexp='yes'>ainda|cedo|tarde|adiante|abaixo|</token>
             <token regexp='yes'>.|,|C.+|SPS00</token>
             <example>Já gostava de chocolate - e agora, mais ainda.</example>
         </antipattern>
-        <antipattern>
-            <token>,</token>
-            <token>mais</token>
-            <token regexp='yes'>esperança|</token>
-            <example>Já gostava de chocolate - e agora, mais ainda.</example>
-        </antipattern>
```
[the remark this answers is not in the document]
Comment (St-ac-y) [reply languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml]: I am still not sure why this AP was deleted.

[the remark this answers is not in the document]
Comment (susanaboatto) [reply languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml]: This was a draft I failed to delete - I fixed this error with another antipattern

**Pull request closed without merging**